server/models/Reservation.js

```diff
@@ -19,9 +19,13 @@ const reservationSchema = new Schema(
       type: String,
       required: true
     },
+    //A room ID to reference the room reserved.
     room: {
-      type: Schema.Types.ObjectId,
-      ref: 'Room'
+      type: String,
+    },
+    //Not needed
+    discount: {
```

Comment:
I think you could just go ahead and delete the discount property since it is not needed